src/main/kotlin/io/perfometer/dsl/HttpDsl.kt

```diff
@@ -93,14 +38,21 @@ class HttpDsl(
         }
     }
 
-    private suspend fun request(
-        httpMethod: HttpMethod,
-        urlString: String?,
-        builder: RequestDsl.() -> Unit
+    fun basicAuth(user: String, password: String) {
+        val credentialsEncoded = Base64.getEncoder().encodeToString("$user:$password".toByteArray())
+        headers(HttpHeaders.AUTHORIZATION to "Basic $credentialsEncoded")
+    }
+
+    suspend fun parallel(
+        builder: suspend HttpDsl.() -> Unit,
     ) {
-        val requestUrl = urlString?.toUrl() ?: baseURL
-        val request = RequestDsl(requestUrl, httpMethod, headers).apply(builder).build()
-        scenarioRunner.runStep(RequestStep(request))
+        runner.runStep(ParallelStep {
```

Comment:
Do we really need to still have ScenarioRunner here? I think we should do this inside ScenarioRunner and drop dependency of HttpDsl to ScenarioRunner.